src/client/containers/QuizPage/QuizPage.container.js

```diff
@@ -1,75 +1,82 @@
-import image2 from '../../assets/images/questionBackgrounds/question2background.png';
-import image3 from '../../assets/images/questionBackgrounds/question3background.png';
-import image4 from '../../assets/images/questionBackgrounds/question4background.png';
-import image5 from '../../assets/images/questionBackgrounds/question5background.png';
+import {  Link} from 'react-router-dom';
+
 import './QuizPage.styles.css';
 
-const QuizQuestions = [
-  {
-    question: 'When visiting a website,what is that you are most intrested in?',
-    image: image1,
-    isAgreementQuestion: false,
-    firstAnswer: 'How the website looks and how easy it is for the users',
-    secondAnswer: 'Logic behind how the website is built',
-    level: 20,
-  },
-  {
-    question:
-      'I can easily notice changes around me including people,culture,trends,etc.',
-    image: image2,
-    isAgreementQuestion: true,
-    firstAnswer: '',
-    secondAnswer: '',
-    level: 40,
-  },
-  {
-    question:
-      'When I have a list of tasks to do,I prefer to multitask rather than focusing on singular tasks one at a time',
-    image: image3,
-    isAgreementQuestion: true,
-    firstAnswer: '',
-    secondAnswer: '',
-    level: 60,
-  },
-  {
-    question: 'I can easily understan what someone else is going through',
-    image: image4,
-    isAgreementQuestion: true,
-    firstAnswer: '',
-    secondAnswer: '',
-    level: 80,
-  },
-  {
-    question: 'I work well under pressure',
-    image: image5,
-    isAgreementQuestion: true,
-    firstAnswer: '',
-    secondAnswer: '',
-    level: 100,
-  },
-];
 export const QuizPage = () => {
   const [currentOn, setCurrentOn] = useState(0);
-  const handleQuestions = () => {
-    setCurrentOn((prev) => prev + 1);
+  const [isLoading, setIsLoading] = useState(false);
+  const [questions, setQuestions] = useState(undefined);
+  const [error, setError] = useState(undefined);
+  const [selectedAnswer, setSelectedAnswer] = useState(undefined);
+
+  const handleQuestions = async () => {
+    let userId = localStorage.getItem('anonymousUserId');
+    if (!userId) {
+      const response = await fetch('/api/users', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({}),
+      });
+      if (response.ok) {
+        const user = await response.json();
+
+        userId = user.userId;
+        localStorage.setItem('anonymousUserId', userId);
+      } else {
+        setError(await response.text());
+      }
+    }
+    if (selectedAnswer !== undefined) {
+      const response = await fetch('/api/quiz-results', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          fk_answer_id: selectedAnswer,
+          fk_user_id: Number(userId),
+        }),
+      });
+      if (response.ok) {
+        setCurrentOn((prev) => prev + 1);
+        setSelectedAnswer(undefined);
```

Comment:
Why don't you use false like you did in line 14?

Reply:
This is because it is not a boolean. We need to reset selectedAnswer back to an undefined state, ready for the users selection on the next answer. The one on line 14 is a boolean.